echopype/mask/api.py

```diff
@@ -161,6 +173,18 @@ def _validate_and_collect_mask_input(
             # open up DataArray using mask path
             mask = xr.open_dataarray(mask, engine=file_type, chunks={}, **storage_options_mask)
 
+        # **Validate boolean-like values - reject NaN**
+        # First check if there are any NaN values
+        if np.any(np.isnan(mask.values)):
+            raise TypeError("Mask must be boolean")
+
+        # Get unique values and check if they are boolean-like
+        unique_vals = np.unique(mask.values)
+
+        # Check if all values are boolean-like (0, 1, True, False)
+        if not np.all(np.isin(unique_vals, [0, 1, True, False])):
+            raise TypeError("Mask must be boolean")
+
```

Comment:
Not specific to the changes you made, but this entire `else` block seems completely redundant. Can you absorb this into the `if isinstance(mask, list)` section by simply making a single input mask a list and just use the code for list?